core/templates/pages/topic-editor-page/topic-editor-page.component.ts

```diff
@@ -178,20 +188,96 @@ angular.module('oppia').directive('topicEditorPage', [
             return validationIssuesCount + prepublishValidationIssuesCount;
           };
 
+          ctrl.onClosingTopicEditorBrowserTab = function() {
+            const topic = TopicEditorStateService.getTopic();
+
+            const topicEditorBrowserTabsInfo: EntityEditorBrowserTabsInfo = (
+              LocalStorageService.getEntityEditorBrowserTabsInfo(
+                EntityEditorBrowserTabsInfoDomainConstants
+                  .OPENED_TOPIC_EDITOR_BROWSER_TABS, topic.getId()));
+
+            if (topicEditorBrowserTabsInfo.doesSomeTabHaveUnsavedChanges() &&
+                UndoRedoService.getChangeCount() > 0) {
+              topicEditorBrowserTabsInfo.setSomeTabHasUnsavedChanges(false);
+            }
+            topicEditorBrowserTabsInfo.decrementNumberOfOpenedTabs();
+
+            LocalStorageService.updateEntityEditorBrowserTabsInfo(
+              topicEditorBrowserTabsInfo,
+              EntityEditorBrowserTabsInfoDomainConstants
+                .OPENED_TOPIC_EDITOR_BROWSER_TABS);
+          };
+
+          let createTopicEditorBrowserTabsInfo = function() {
```

Comment:
Prefer const here and for the two methods below.

Reply:
Done.